src/SALib/analyze/hdmr.py

```diff
@@ -0,0 +1,692 @@
+from __future__ import division
+from __future__ import print_function
+import itertools
+import time
+import numpy as np
+from SALib.plotting import hdmr
+from scipy import (stats, special, interpolate)
+from . import common_args
+from ..util import read_param_file, ResultDict
+from matplotlib import pyplot as plt
+
+
+def analyze(problem, X, Y, options={}):
```

Comment:
Please follow numpydoc standards like this:

```python
def function(argument1, argument2=None):
    """One line description of the function

    Empty line above, and now this can be a longer paragraph describing the function

    Arguments
    ---------
    argument1 : int
        Description of argument1
    argument : float, default=None
        Description of argument2

    Returns
    -------
    float
        Description of the return value and type

    Notes
    -----
    Other notes etc.  For more information, see https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard